torch/nested/_internal/ops.py

```diff
@@ -215,10 +309,7 @@ def linear_default(func, *args, **kwargs):
     weight = new_kwargs["weight"]
     bias = new_kwargs["bias"]
 
-    new_values = torch.mm(inp._values, weight)
-    if bias is not None:
-        new_values += bias
-    return NestedTensor(new_values, **extract_kwargs(inp))
+    return NestedTensor(func(inp._values, **new_kwargs), **extract_kwargs(inp))
```

Comment:
@soulitzer FYI this change fixes linear to expect `weight` in the form of `(out_channels, in_channels)`. Backward and tests have to change to accommodate this as well

Reply:
whoops, thanks!